### PR TITLE
Integrate Blog page with Netlify CMS data

### LINKED ISSUE #68: Integrate Blog page with Netlify CMS data

### PR DESCRIPTION
**Pull request by mecies**
### Description

This PR covers:
- Update BlogPost component with onClick
- Update BlogPostDialog component with action handlers
- Refactor repetitive logic to determine the components type to a reusable hook
- Fetch data to blog-post page from CMS and add a missing image field in information structure
- Update stories for changed components

Closes #68  

### Preview
Screenshot 2020-09-09 at 22 57 23
Screenshot 2020-09-09 at 22 57 31
Screenshot 2020-09-09 at 22 57 35

### CODE REVIEW

File: src/pages/blog-page/index.tsx [hunk 1]
```diff
@@ -121,21 +127,66 @@ const useStyles = makeStyles((theme) => ({
...
       <Box className={classes.main}>
         <Box className={classes.navbar}>NavBar</Box>
         <Box className={classes.mainContent}>
           <Box className={classes.blogContainer}>
             <SectionTitle className={classes.title}>Blog</SectionTitle>
             <Box className={classes.blogPosts}>
-              {[...Array(4)].map((_, index) => (
-                <BlogPost key={index} className={classes.blogPost} {...blogData} />
+              {blogData.blog_post.map((blogPost, index) => (
+                <div key={index}>
```
Comment (KonradSzwarc) [inline src/pages/blog-page/index.tsx]: Don't use the index as key

File: src/hooks/useComponentType.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { useMediaQuery, useTheme } from '@material-ui/core';
+
+export const useComponentType = () => {
+  const theme = useTheme();
+  const isMobile = useMediaQuery(theme.breakpoints.down('xs'));
+  const isTablet = useMediaQuery(theme.breakpoints.down('md'));
+  const isDesktop = !isMobile && !isTablet;
+  const componentType: 'mobile' | 'tablet' | 'desktop' = isMobile ? 'mobile' : isTablet ? 'tablet' : 'desktop';
+  return { componentType, isMobile, isTablet, isDesktop };
```
Comment (KonradSzwarc) [inline src/hooks/useComponentType.ts:9]: Do we need to return `isMobile`, `isTablet`, and `isDasktop` if we can use `componentType` to determine the size

File: src/components/BlogPost/BlogPostDialogTablet.tsx [hunk 3]
```diff
@@ -37,24 +37,24 @@ const useStyles = makeStyles((theme) => ({
...
         <Box position="absolute" left="620px">
           <Box mb={5}>
-            <IconButton {...props} size={'small'}>
+            <IconButton onClick={props.handleClose} size={'small'}>
               <X size={30} />
             </IconButton>
           </Box>
           <Box mb={2.5}>
-            <IconButton {...props} size={'small'}>
+            <IconButton onClick={props.handlePrev} size={'small'}>
```
Comment (KonradSzwarc) [inline src/components/BlogPost/BlogPostDialogTablet.tsx]: `size="small"`

**Review by KonradSzwarc — CHANGES_REQUESTED** [review]

File: src/hooks/useComponentType.ts (hunk 2 above)
Comment (mecies) [reply src/hooks/useComponentType.ts:9]: I think that it is more elegant to return them.
I changed it as such in the last commit, without it everyone would have to do the check `componentType === "mobile"` etc and I think that this way is a better solution.

**Review by KlaraGajaszek — APPROVED** [review]

File: src/components/BlogPost/BlogPostDialogTablet.tsx (hunk 3 above)
Comment (mecies) [reply src/components/BlogPost/BlogPostDialogTablet.tsx]: Good catch :D Will replace project-wide

File: src/pages/blog-page/index.tsx (hunk 1 above)
Comment (mecies) [reply src/pages/blog-page/index.tsx]: Good catch, will replace in all pages aswell

**Review by KonradSzwarc — APPROVED** [review]

**Pull request merged**